tests/test_everflow_testbed.py

```diff
@@ -0,0 +1,368 @@
+import os
+import time
+from ansible_host import AnsibleHost
+import pytest
+from ptf_runner import ptf_runner
+from abc import ABCMeta, abstractmethod
+import ipaddr as ipaddress
+
+BASE_DIR = os.path.dirname(os.path.realpath(__file__))
+DUT_TMP_DIR = os.path.join('tmp', os.path.basename(BASE_DIR))
+FILES_DIR = os.path.join(BASE_DIR, 'files')
+TEMPLATE_DIR = os.path.join(BASE_DIR, 'templates')
+
+EVERFLOWE_TABLE_RULE_CREATE_TEMPLATE = 'acl_rule_persistent.json.j2'
+EVERFLOWE_TABLE_RULE_CREATE_FILE = 'acl_rule_persistent.json'
+EVERFLOWE_TABLE_RULE_DELETE_FILE = 'acl_rule_persistent-del.json'
+DUT_RUN_DIR = '/home/admin/everflow_tests'
+
+@pytest.fixture(scope='module')
+def setup_info(duthost, testbed):
+    """
+    setup fixture gathers all test required information from DUT facts and testbed
+    :param duthost: DUT host object
+    :param testbed: Testbed object
+    :return: dictionary with all test required information
+    """
+
+    tor_ports = []
+    spine_ports = []
+    port_channels = []
+    everflow_table_ports = []
+
+    # gather ansible facts
+    mg_facts = duthost.minigraph_facts(host=duthost.hostname)['ansible_facts']
+    #gather switch capability facts
+    switch_capability_facts = duthost.switch_capabilities_facts()['ansible_facts']
+    #gather host facts
+    host_facts = duthost.setup()['ansible_facts']
+
+    # get the list of TOR/SPINE ports
+    for dut_port, neigh in mg_facts['minigraph_neighbors'].items():
+        if 'T0' in neigh['name']:
+            tor_ports.append(dut_port)
+        elif 'T2' in neigh['name']:
+            spine_ports.append(dut_port)
+
+    # get the list of port channels
+    port_channels = mg_facts['minigraph_portchannels']
+
+    # get the list of port to be combined to ACL tables
+    everflow_table_ports += tor_ports
+    if testbed['topo']['name'] in ('t1-lag', 't1-64-lag', 't1-64-lag-clet'):
+        everflow_table_ports += port_channels
+    else:
```

Comment:
Do you need to add spine ports for non-T0 topologies? Else here would include T1 without lag, right?

Reply:
Latest code doesn't have this.  Binding of ACL to port is implictilty done in Sonic so we don't need to do.

Reply:
As we discussed. Please add a comment here.

Reply:
@yxieca Done